dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/client/jose4j/util/Jose4jUtil.java

```diff
@@ -92,40 +100,100 @@ public Jose4jUtil(SSLSupport sslSupport) {
+        String originalIdTokenString = idTokenStr;
+        String accessTokenStr = tokens.get(Constants.ACCESS_TOKEN);
+        String refreshTokenStr = tokens.get(Constants.REFRESH_TOKEN);
         String clientId = clientConfig.getClientId();
-        try {
-            if (tokenStr == null || tokenStr.isEmpty()) {
+        Hashtable<String, Object> customProperties = new Hashtable<String, Object>();
+
+        List<String> tokensOrderToFetchCallerClaims = clientConfig.getTokenOrderToFetchCallerClaims();
+        try {          
+            if ((idTokenStr == null || idTokenStr.isEmpty()) && tokensOrderToFetchCallerClaims.size() == 1) {
                 // This is for ID Token only
                 Tr.error(tc, "OIDC_CLIENT_IDTOKEN_REQUEST_FAILURE", new Object[] { clientId, clientConfig.getTokenEndpointUrl() });
                 return new ProviderAuthenticationResult(AuthResult.SEND_401, HttpServletResponse.SC_UNAUTHORIZED);
             }
-            JwtContext jwtContext = validateJwtStructureAndGetContext(tokenStr, clientConfig);
-            JwtClaims jwtClaims = parseJwtWithValidation(clientConfig, jwtContext.getJwt(), jwtContext, oidcClientRequest);
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "post jwtClaims: " + jwtClaims + " firstPass jwtClaims=" + jwtContext.getJwtClaims());
-            }
-            OidcTokenImplBase idToken = new OidcTokenImplBase(jwtClaims, accessToken, refreshToken, clientId, oidcClientRequest.getTokenTypeNoSpace());
+            
+            Map<String, JwtClaims> tokenClaimsMap = new HashMap<String, JwtClaims>();
 
-            if (idToken.getSubject() == null) {
+            JwtClaims idTokenClaims = getClaimsFromIdToken(idTokenStr, clientConfig, oidcClientRequest);
+            OidcTokenImplBase idToken = getOidcToken(idTokenClaims, accessTokenStr, refreshTokenStr, clientId, Constants.TOKEN_TYPE_ID_TOKEN);
+            String sub = idToken.getSubject();
+            if (sub == null) {
                 return new ProviderAuthenticationResult(AuthResult.SEND_401, HttpServletResponse.SC_UNAUTHORIZED);
             }
-            AttributeToSubject attributeToSubject = new AttributeToSubject(clientConfig, idToken);
-            if (attributeToSubject.checkUserNameForNull()) {
+            tokenClaimsMap.put(Constants.TOKEN_TYPE_ID_TOKEN, idTokenClaims);
+
+            UserInfoHelper userInfoHelper = new UserInfoHelper(clientConfig, sslSupport);
+            String userInfoStr = userInfoHelper.getUserInfoIfPossible(sub, accessTokenStr, sslSocketFactory, oidcClientRequest);
+            JwtClaims userInfoClaims = null;
+            if (userInfoStr != null) {
+                try {
+                    userInfoClaims = getClaimsFromUserInfo(userInfoStr);
+                    
+                } catch (Exception e) {
+                    Tr.error(tc, "Invalid user info: " + userInfoStr);
```

Comment:
```suggestion
                    Tr.debug(tc, "Invalid user info: " + userInfoStr);
```
The `Tr.error()` method emits messages to the console.log and messages.log that require translation. The `Tr.debug()` method should be used here instead.

Reply:
This is a higher priority issue to fix.